examples/components/CCFRAUD_VERTICAL/traininsilo/contributor.py

```diff
@@ -0,0 +1,404 @@
+        """
+        logger.info(f"Train data dir: {train_data_dir}, Test data dir: {test_data_dir}")
+        if self._global_rank == 0:
+            self.fraud_weight_ = np.loadtxt(train_data_dir + "/fraud_weight.txt").item()
+        train_df = pd.read_csv(train_data_dir + "/data.csv", index_col=0)
+        test_df = pd.read_csv(test_data_dir + "/data.csv", index_col=0)
+        if model_name == "SimpleLinear":
+            train_dataset = datasets.FraudDataset(train_df)
+            test_dataset = datasets.FraudDataset(test_df)
+        else:
+            train_dataset = datasets.FraudTimeDataset(train_df)
+            test_dataset = datasets.FraudTimeDataset(test_df)
+
+        logger.info(
+            f"Train data samples: {len(train_df)}, Test data samples: {len(test_df)}"
+        )
+
+        return train_dataset, test_dataset, train_dataset.features_count()
+
+    def log_params(self, client, run_id):
+        client.log_param(
+            run_id=run_id, key=f"learning_rate {self._experiment_name}", value=self._lr
+        )
+        client.log_param(
+            run_id=run_id, key=f"epochs {self._experiment_name}", value=self._epochs
+        )
+        client.log_param(
+            run_id=run_id,
+            key=f"batch_size {self._experiment_name}",
+            value=self._batch_size,
+        )
+        client.log_param(
+            run_id=run_id,
+            key=f"loss {self._experiment_name}",
+            value=self.criterion_.__class__.__name__,
+        )
+        client.log_param(
+            run_id=run_id,
+            key=f"optimizer {self._experiment_name}",
+            value=self.optimizer_.__class__.__name__,
+        )
+
+    def log_metrics(self, client, run_id, key, value, pipeline_level=False):
+        if pipeline_level:
+            client.log_metric(
+                run_id=run_id,
+                key=f"{self._experiment_name}/{key}",
+                value=value,
+            )
+        else:
+            client.log_metric(
+                run_id=run_id,
+                key=f"{self._iteration_name}/{self._experiment_name}/{key}",
+                value=value,
+            )
+
+    def local_train(self, checkpoint):
+        """Perform local training for a given number of epochs
+
+        Args:
+            checkpoint: Previous model checkpoint from where training has to be started.
+        """
+
+        if checkpoint:
+            self.model_.load_state_dict(torch.load(checkpoint + "/model.pt"))
+
+        with mlflow.start_run() as mlflow_run:
+            # get Mlflow client and root run id
+            mlflow_client = mlflow.tracking.client.MlflowClient()
+            root_run_id = os.environ.get("AZUREML_ROOT_RUN_ID")
+            logger.debug(f"Root runId: {root_run_id}")
+
+            # log params
+            self.log_params(mlflow_client, root_run_id)
+
+            logger.debug("Local training started")
+
+            train_metrics = RunningMetrics(
```

Comment:
do we even need that here? looks like the only thing we do is to reset those metrics